vyper/codegen/function_definitions/internal_function.py

```diff
@@ -41,8 +41,8 @@ def generate_ir_for_internal_function(
 
     for arg in func_t.arguments:
         # allocate a variable for every arg, setting mutability
-        # to False to comply with vyper semantics, function arguments are immutable
-        context.new_variable(arg.name, arg.typ, is_mutable=False)
+        # to True to allow internal function arguments to be mutable
```

Comment:
is the `is_mutable` flag on `VariableRecord` even still used? or is it dead

Reply:
It is still used. This change was triggered by this line:
https://github.com/vyperlang/vyper/blob/7f18aeee59abbf3f4657edc94a8b354731cce19b/vyper/codegen/stmt.py#L395

Reply:
hmmm that seems like it could be dead code, that kind of thing should be caught in the frontend.

Reply:
but maybe it is good to have the extra check during codegen

Reply:
i wonder if we can actually remove the VariableRecord thing entirely though, it's pretty redundant with VariableInfo in the frontend.

Reply:
closing this thread for now, will revisit the VariableRecord thing later